application/usecases/sendRolesDropdownMessage/sendRolesDropdownMessageUseCase.ts

```diff
@@ -0,0 +1,79 @@
+import { ActionRowBuilder, StringSelectMenuBuilder } from "discord.js";
+import ROLES_MESSAGES_MAP from "../../../assets/consts/rolesMap";
+import { RoleInterface } from "../../../assets/interfaces/roleInterface";
+import { CustomEmoji } from "../../../domain/interface/customEmoji.interface";
+import ChatService from "../../../domain/service/chatService";
+import { SendRolesDropdownMessageInput } from "./sendRolesDropdownMessageInput";
+
+export default class SendRolesDropdownMessageUseCase {
+  private chatService: ChatService;
+
+  constructor({ chatService }: { chatService: ChatService }) {
+    this.chatService = chatService;
+  }
+
+  async execute({ channelId, guildId, memberId }: SendRolesDropdownMessageInput) {
+    try {
+      if (!guildId || !memberId) return;
+
+      // Verify is ther member is a admin or moderator
+      const allowedRoles = ["Admin", "Moderador"];
+
+      const hasPermission = await this.chatService.isUserWithRoleName(guildId, memberId, allowedRoles);
+
+      if (!hasPermission) {
+        console.log(`User does not have permission to execute this command '!roles'`);
+        return;
+      }
+
+      const emojis: CustomEmoji[] | undefined = (await this.chatService.getGuildEmojis(guildId)) || undefined;
+
+      Object.values(ROLES_MESSAGES_MAP).forEach(async (roleMessage) => {
+        const message = SendRolesDropdownMessageUseCase.getRolesDropdownMessage(roleMessage, emojis);
+
+        await this.chatService.sendMessageToChannel(message, channelId);
+      });
```

Comment:
Temo que este `forEach` com os `await` possam não estar a fazer o que queres, pelo menos pela ordem certa. Ao invocares como invocaste aqui ele não respeita a ordem e faz-te diversas chamadas assíncronas para o envio de mensagens.

[O `for...const` penso que te ajude aqui](https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop).